dali/pipeline/operator/sequence_operator.h

```diff
@@ -0,0 +1,595 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_PIPELINE_OPERATOR_SEQUENCE_OPERATOR_H_
+#define DALI_PIPELINE_OPERATOR_SEQUENCE_OPERATOR_H_
+
+#include <memory>
+#include <string>
+#include <type_traits>
+#include <utility>
+#include <vector>
+#include "dali/pipeline/data/tensor.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/argument.h"
+#include "dali/pipeline/operator/common.h"
+#include "dali/pipeline/operator/op_spec.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/operator/sequence_shape.h"
+
+namespace dali {
+
+namespace detail {
+inline bool is_per_frame(const TensorVector<CPUBackend> &arg_input) {
+  const auto &layout = arg_input.GetLayout();
+  return layout.size() > 0 && layout[0] == 'F';
+}
+}  // namespace detail
+
+/**
+ * @brief SequenceOperator
+ * Provides generic support for sequence processing to an operator by unfolding
+ * a batch of sequences into a batch of frames. Applicability of the SequenceOperator
+ * is limited to operators that can process frames independently. For example, an operator
+ * that applies convolution to a batch of images can be turned into the operator that supports
+ * batches of sequences of frames, given that there is no need to convolve along frames dimension.
+ *
+ * Adds support for per-frame tensor arguments: if a tensor argument is marked in the schema as
+ * supporting per-frame tensor values and the tensor argument layout starts with `F`, the
+ * outermost dimension of the tensor argument will be unfolded to match the the expanded
+ * input of the operator.
```

Comment:
Is this reflected in the operator documentation? I mean, the one that user is actually reading?

Reply:
The follow-up of this task is https://github.com/NVIDIA/DALI/pull/3723